data/appimagetool

```diff
@@ -1,4 +1,4 @@
-https://github.com/AppImage/AppImageKit/releases/download/continuous/appimagetool-x86_64.AppImage
+https://github.com/AppImage/AppImageKit
```

Comment:
This won't work since we have multiple AppImages within the release

Reply:
It's pretty easy to search for the AppImage based on its name when the user requests to install/update/etc that AppImage.  I'm already doing this by getting rid of all of the extra info on the end of links that you provide and searching for the user inputted AppImage name in Github's API for that repo's releases.

I imagine that if I'm able to do this reliably in a bash script, it should be relatively easy to do in other more sophisticated programming languages.

Reply:
I'm (currently) using bash, too...

Reply:
This is more a logic question: From where if not from that URL should the system know which AppImage we are referring to?

Reply:
All you have to do is add a grep for the AppImage you wish to find from GIthub's API.  This is how I manage it:
```
APPIMAGE_NAME="$(grep -iv '.*ia32*.\|.*i686*.\|.*i386*.' "$APPIMAGE_INFO" | grep -i "$INSTIMG" | grep -im 1 '"name":*..*AppImage"' | cut -f4 -d'"')"
NEW_APPIMAGE_VERSION="$(grep -iv '.*ia32*.\|.*i686*.\|.*i386*.' "$APPIMAGE_INFO" | grep -B 1 -im 1 '"browser_download_url":*..*AppImage"' | head -n 1 | cut -f4 -d'"')"
GITHUB_APPIMAGE_URL="$(grep -iv '.*ia32*.\|.*i686*.\|.*i386*.' "$APPIMAGE_INFO" | grep -i "$INSTIMG" | grep -im 1 '"browser_download_url":*..*AppImage"' | cut -f4 -d'"')"
```
Where `$INSTIMG` is the AppImage that is either being installed, checked for updates, or upgraded.

The first grep gets rid of any data that is not for 64bit systems.  The second grep makes sure that we only find the specific AppImage that is being downloaded/updated/upgraded.  The final grep finds the json key that stores the information that is needed for downloading/updating/upgrading the AppImage.

This doesn't work for all AppImages as some people do not have consistent names between what they call their application and what they name their packages, so if the first set of greps does not find an AppImage, the second set below is used as a fallback to find any AppImage that is listed in the repo.

```
if [ -z "$APPIMAGE_NAME" ]; then
    APPIMAGE_NAME="$(grep -iv '.*ia32*.\|.*i686*.\|.*i386*.' "$APPIMAGE_INFO" | grep -im 1 '"name":*..*AppImage"' | cut -f4 -d'"')"
    NEW_APPIMAGE_VERSION="$(grep -iv '.*ia32*.\|.*i686*.\|.*i386*.' "$APPIMAGE_INFO" | grep -B 1 -im 1 '"browser_download_url":*..*AppImage"' | head -n 1 | cut -f4 -d'"')"
    GITHUB_APPIMAGE_URL="$(grep -iv '.*ia32*.\|.*i686*.\|.*i386*.' "$APPIMAGE_INFO" | grep -im 1 '"browser_download_url":*..*AppImage"' | cut -f4 -d'"')"
fi
```

Searching Github's API with greping the AppImage name works very well for finding specific AppImages that are hosted in repos with multiple AppImage downloads available.

For example, I am able to reliably find the appimagetool AppImage download from AppImageKit's releases page reliably without ever finding the AppImage that I don't want:
![image](https://user-images.githubusercontent.com/16279904/29849290-63310bc2-8ceb-11e7-81af-e8ea5f308cd0.png)

Reply:
> add a grep for the AppImage you wish to find from GIthub's API

Question is, what to grep for. That is what we lose by changing "https://github.com/AppImage/AppImageKit/releases/download/continuous/appimagetool-x86_64.AppImage" to "https://github.com/AppImage/AppImageKit/". Of course we could invent something like "https://github.com/AppImage/AppImageKit/#appimagetool-x86_64.AppImage"...

Reply:
We know what to grep for by looking at the file names /data from AppImageHub.

Reply:
The `/data` directory is not meant to be used by anything else but AppImageHub internally. We will rename it to `/input` soon, to make this distinction clearer. We will also make an `/output` directory that will contain, e.g., JSON files that are meant for consumption by others.

Reply:
Either way, I assume you'll be storing the names of AppImages that are listed on AppImageHub somewhere there, and developers looking to use that resource will be able to use that list in the same manner that I'm using the /data directory currently.

I'm currently hosting 3 different AppImages on a repo of my own, and I'm able to easily find the download and update information for those 3 different AppImages on the same repo.  

As long as there is a list that contains the program's name and the URL to the github repo the AppImage is hosted on, finding multiple AppImages on the same repo will not be a problem.

Reply:
Please at the very least merge the changes that you don't think cause problems or fix the broken links yourself.  Having a broken link for nteract and having API links in the list makes it more difficult to interact with.

Reply:
Again, we really need to define an output format. The `data` directory is purely _input_ (prior to review), and may contain broken stuff all the time. At this moment, the closest thing to an output format are the files in https://github.com/AppImage/appimage.github.io/tree/master/apps, but be aware that they are still subject to change.

Reply:
Honestly, those .md files are not really usable for me at all.  The list in /data is much more usable.

Reply:
Please at the very least do *not* remove /data.  /apps is not in a format that I find workable.

Reply:
Providing this information in JSON should be easily doable in this repository using Jekyll. PR welcome.

Reply:
Nevermind, I guess I'll just convert those pages to a workable format myself.

Reply:
It's _really_ easy in Jekyll...

Reply:
It took me a bit to realize those were even in yaml format.  Why you've put them in markdown is beyond me...

Reply:
Still learning Jekyll...

Reply:
Just put the raw yaml files up.  Putting them in markdown just adds an extra step to make them usable.

Reply:
I've found a [bash function for parsing yaml](https://stackoverflow.com/a/21189044) that someone posted, and so far it's working very well for the few files I've thrown threw it from the /apps dir here.  I'm even able to do it without downloading the files by piping wget through the function, so that works out well.

It would have been nice if those files were just in raw yaml format instead of markdown to avoid confusion, but either way, sorry for questioning your methods here.  Keep up the good work; looking forward to having the rest of the AppImage's info in the /apps dir so I can start taking advantage of it!

Reply:
If you find a way to have this work as pure yaml without md front matter, please send a Pull Request.